cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/default-service/AwsDefaults.service.tsx

```diff
@@ -11,4 +11,12 @@ export class AwsDefaults {
     static selectDefalutEC2KeyPairs = (keyPairs: AWSKeyPair[]) => {
         return keyPairs && keyPairs.length > 0 ? keyPairs[0].name : undefined;
     };
+
+    static setDefaultNodeNodeType = (nodeTypeList: string[], nodeProfile: string) => {
+        if (nodeTypeList.indexOf(validateDefaultNodeType(nodeProfile)) > -1) {
+            return validateDefaultNodeType(nodeProfile);
+        } else {
+            return nodeTypeList[nodeTypeList.length / 2];
```

Comment:
Please add a TODO comment here because we need to come up with a more accurate solution for selecting a nodeType in this 'else' case. I will continue to work on figuring out what the logic should be.

Something like: TODO: refactor to select optimal nodeType when preferred default not found

Reply:
Got it.